src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -245,6 +349,95 @@ private Exception GetCurlException(int code)
             return new Exception(GetCurlErrorString(code));
         }
 
+        private void SetCurlOption(SafeCurlHandle handle, int option, object value)
+        {
+            int result;
+
+            // TODO: For some reason using generic is not working here
+            //       SetCurlOption<TOption>(SafeCurlHandle, int, TOption) compilation fails
+            if (value is string)
+            {
+                result = Interop.libcurl.curl_easy_setopt(handle, option, (string)value);
+            }
+            else if (value is long)
+            {
+                result = Interop.libcurl.curl_easy_setopt(handle, option, (long)value);
+            }
+            else if (value is Interop.libcurl.curl_readwrite_callback)
+            {
+                result = Interop.libcurl.curl_easy_setopt(handle, option, (Interop.libcurl.curl_readwrite_callback)value);
+            }
+            else if (value is Interop.libcurl.curl_ioctl_callback)
+            {
+                result = Interop.libcurl.curl_easy_setopt(handle, option, (Interop.libcurl.curl_ioctl_callback)value);
+            }
+            else
+            {
+                result = Interop.libcurl.curl_easy_setopt(handle, option, (IntPtr) value);
+            }
+
+            if (result != CURLcode.CURLE_OK)
+            {
+                throw new HttpRequestException(SR.net_http_client_execution_error, GetCurlException(result));
+            }
+        }
+
+        private static bool InitializeCurl()
+        {
+            // TODO: A process-wide curl_global_init is required. According to the documentation,
```

Comment:
T1 will still call libcurl APIs via CurlHandler but the thread got created before curl was initialized by T2. Actually I realized that CurlHandler implementation is such that the handle is used in a thread that was created after init. Perhaps this was what you meant as well. I am removing the TODO since I think we are covered by having a static ctor